### ISSUE TITLE
Cannot use two TabbedForm at the same time

### ISSUE THREAD

**Issue by Mattia-Longhin** [marmelab/react-admin#4072]
Hello Marmelab, 
Thank you for your work!

**TLDR**
_`TabbedForm` uses browser url to handle their state. This does not work when we have two `TabbedForm` at the same time. (example: one in the main view and one in a dialog)._

**Longer version**

I'm reproducing the QuickCreateButton functionality of your blog post [React Admin Advanced Recipes - Adding Related Records Inline With Custom Forms](https://marmelab.com/blog/2018/08/27/react-admin-tutorials-custom-forms-related-records.html).

It works like a charm with `SimpleForm` but breaks with `TabbedForm`.

I have a main `TabbedForm` with a `ReferenceInput`. Referenced resource can be created via a _CreateNew_ button. This opens a dialog with a secondary `TabbedForm`.

**What you were expecting:**

I'm expecting the `TabbedForm` in the dialog to be independent from the main `TabbedForm`. 

**What happened instead:**

When both `TabbedForm`s are displayed, they both try to use browser location to determine which Tab is shown. This breaks the behaviour of both `TabbedForm`s.

**Steps to reproduce:**

I put together a [Codesandbox](https://codesandbox.io/s/quizzical-framework-zbupy?fontsize=14&hidenavigation=1&theme=dark) to show the issue.

- Open the Create form for the Contacts resource
- Click on Create button below Company input
- Click on a tab in the dialog that opens

**Other information:**

I was able to fix this by creating a local copy of the TabbedForm and creating a local state for each TabbedForm instead of the shared url based logic. This works but having a local copy of part of the library is too hacky for customer projects.

**Environment**

* React-admin version: v3.0.1 (older versions have the same issue)

**Comment by fzaninotto** [comment]
Thanks for the report. 

That's a known limitation of `TabbedForm`. You'll have to use a custom form in your dialog.

**Comment by legendstar99999** [comment]
Good
